src/js/Sort/CellFormatter.js

```diff
@@ -174,6 +174,24 @@ function ftDateTimeToNumber(text) {
 	return isNaN(dateObj.getTime()) ? text : dateObj.getTime();
 }
 
+/**
+ * Parses a datetime string as a number for sorting. -- https://html.spec.whatwg.org/multipage/text-level-semantics.html#attr-time-datetime
+ * @example
+ *  dateTimeToNumber('11-17') //UNIX epoch, assumes current year
+ *  dateTimeToNumber('2011-11-18') //UNIX epoch
+ *  dateTimeToNumber('2011-11') //UNIX epoch, first of month
+ *  dateTimeToNumber('2011-11-18T14:54') //UNIX epoch including time
+ *  dateTimeToNumber('14.54') //14.54
+ *  dateTimeToNumber('Not a known date') //Note a known date
+ * @param {String} text The datetime string to operate on
+ * @access private
+ * @returns {Number|String} Number representation of datetime string or the original datetime string if it could not be turned into a number.
+ */
+function dateTimeToNumber(text) {
+	const dateObj = new Date(text);
```

Comment:
I think whilst working on `o-table` I dismissed `Date` because of this MDN warning:

>It is not recommended to use Date.parse as until ES5, parsing of strings was entirely implementation dependent. There are still many differences in how different hosts parse date strings, therefore date strings should be manually parsed (a library can help if many different formats are to be accommodated).

But how silly, we live in the future now! 🙈